src/interfaces/customer.interface.ts

```diff
@@ -0,0 +1,11 @@
+import { Role } from '../enums/role.enum';
+
+export interface ICustomer {
```

Comment:
**suggestion(non-blocking):** now as a personal preference, I like to keep all interfaces, enums and other types grouped under the same `types` folder

Reply:
Glad to hear! I was actually wondering how should I approach handling the domain interfaces and the derived types ( for  handling things like placing orders, updating products, etc) and my idea was to separate them completely in interfaces (domain) and types(the derived types). Fixed :)